### ISSUE TITLE
Possible bug: PEC boundary in 1D

### ISSUE THREAD

**Issue by snicks11** [ECP-WarpX/WarpX#3664]
I am using a 1D uniform magnetized plasma with reflecting boundaries for particles and Dirichlet (PEC) boundaries for fields with an EM solver. I am not explicitly setting the boundary potentials; I understand that they are set to zero by default, which is what I want. As I understand, the PEC boundaries should be clamping the tangential (x and y) electric fields to zero. In fact, what I see is that only Ey is clamped to zero. (The PEC boundary also sets the normal component of magnetic fields (Bz) to zero, but in 1D this is automatically satisfied.) Possibly as a consequence, I am seeing an unphysical instability occur, with both total particle and field energy increasing when there are no physical sources of free energy in the plasma. Evidential plots are below, as well as the input file. The input file is also included in a zip folder. 

A plot of Ey vs z at the final timestep. The field is correctly clamped to zero at the boundaries.
![image](https://user-images.githubusercontent.com/102836570/216173261-1a2f6add-ac55-450f-8a8a-77ab9c50c14c.png)

A plot of Ex vs z at the final timestep. The field is not clamped to zero at the boundaries. 
![image](https://user-images.githubusercontent.com/102836570/216173049-0aa2d1ee-4d4a-4aa4-a67c-5ce6aa1aa8da.png)

A colormap of Bx in space and time, showing wave activity. (Time runs downward on the vertical axis, and space runs to the right on the horizontal axis.)
![image](https://user-images.githubusercontent.com/102836570/216172721-379dc196-8157-40ea-92af-cb623c64aef3.png)

A plot of total particle energy vs time.
![image](https://user-images.githubusercontent.com/102836570/216173648-05dd2c3b-31f4-4b3c-90f4-ffa3a0725ea3.png)

A plot of total field energy vs time. This plot and the plot above show the presence of an instability. 
![image](https://user-images.githubusercontent.com/102836570/216173736-a2fe7780-aebb-4311-9013-b144f74170e2.png)

PICMI-translated input deck
![image](https://user-images.githubusercontent.com/102836570/216172335-3bde39a8-1196-42eb-8f9a-ee1eedbcc8f2.png)
[inputs_1d_picmi.zip](https://github.com/ECP-WarpX/WarpX/files/10561982/inputs_1d_picmi.zip)

**Comment by ax3l** [comment]
For the first issue, 1D
> In fact, what I see is that only Ey is clamped to zero.

@RevathiJambunathan @prkkumar could you take a look if the PEC boundaries in 1D need a fix?

**Comment by prkkumar** [comment]
Thanks for reporting this issue, @snicks11 ! There is indeed a bug in 1D PEC. The way we calculate which component is tangential and which is normal has problem for the 1D case. 

Its done here https://github.com/ECP-WarpX/WarpX/blob/b29d4e55ed4633696e3ceaf505df0d56e1324d43/Source/BoundaryConditions/WarpX_PEC.H#L125

Both x and y components are tangential to z boundary (only boundary in 1D). Another conditional for the 1D case after line 123 should fix it.

I made this fix locally and tested it with your input script. Here is electric field at time step 100, before and after the fix. 

**zoomed near lower boundary**
![image](https://user-images.githubusercontent.com/89051199/216464146-a850b22e-f2c4-46cb-844b-e3b6812388e4.png)

**zoomed near upper boundary**
![image (1)](https://user-images.githubusercontent.com/89051199/216464208-4072baf7-2427-47a4-83fa-7d3dfe45d74a.png)

PR with the fix coming up shortly!

**Comment by ax3l** [comment]
Thanks a lot, @prkkumar.
I'll still merge this into the `23.02` release :)

**Comment by snicks11** [comment]
@ax3l @prkkumar 
Hi all, sorry to potentially re-open this issue. I'd first like to say that I indeed see that PEC boundaries are indeed working better. However, with further testing I am also seeing an energy leak into the simulation for both fields and particles. (My input deck is essentially the same as before, but for the purpose of reproduction I have attached it below.) This energy leak seems to be related to the formation of a sheath (Ez) at the ends of the domain. Of course, PEC boundaries allow a nonzero Ez, but with reflecting particles, this sheath is unexpected. If I use particles of equal mass, the sheath and energy leak are suppressed. Also using a large number of particles per cell seems to help. 

Energy in transverse field components:
![image](https://user-images.githubusercontent.com/102836570/217348418-b6abe2a6-f129-43ac-b19d-1f0c8577aadd.png)

Energy in particles:
![image](https://user-images.githubusercontent.com/102836570/217348487-be4737d7-2530-4054-ac98-cce6c1225fbe.png)

Field traces at beginning and end of domain:
![image](https://user-images.githubusercontent.com/102836570/217348197-05f36ef1-1a46-443e-ab0e-6510fd7cc968.png)

Input deck:
[inputs_1d_picmi.zip](https://github.com/ECP-WarpX/WarpX/files/10678829/inputs_1d_picmi.zip)
